src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/StringExtensions.cs

```diff
@@ -24,5 +25,24 @@ public static IEnumerable<string> AsLines(this string content)
                 }
             }
         }
+
+        /// <summary>
+        /// Attempt to parse an Enum value from a string in a case-insensitive manner.
+        /// </summary>
+        /// <typeparam name="TEnum">The type of the enum</typeparam>
+        /// <param name="value">The string value</param>
+        /// <returns>
+        /// A <see cref="Result{TOk,TError}"/> containing the enum value, or a string describing the parse error.
+        /// </returns>
+        public static Result<TEnum, string> ParseEnum<TEnum>(this string value) where TEnum : struct
```

Comment:
Suggest directly constraining to **enum** (permitted since [C# 7.3](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-3)):

```suggestion
        public static Result<TEnum, string> ParseEnum<TEnum>(this string value) where TEnum : Enum
```

Reply:
Done (needs to be `struct` too, because of the constraint on `Enum.TryParse`)